### PR TITLE
feat: bring back deno <script>

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref #2446.

This PR brings back ability to run scripts using `deno <script>` command.

### CODE REVIEW

**Comment by bartlomieju** [comment]
Current status:
```shellsession
$ deno // open REPL, no flags allowed
$ deno --foo // unrecognized arg, exit
$ deno --allow-net script.ts // run script.ts with net perm
$ deno run --allow-net script.ts // run script.ts with net perm
$ deno --allow-read run --allow-net script.ts // run script.ts with net perm
```

Please notice that "read" permission is ignored in last example, I'll try to get it working but it's tricky. 

CC @ry @MarkTiedemann @kitsonk @hayd

File: cli/flags.rs [hunk 1]
```diff
@@ -35,11 +35,61 @@ static ENV_VARIABLES_HELP: &str = "ENVIRONMENT VARIABLES:
     DENO_DIR        Set deno's base directory
     NO_COLOR        Set to disable color";
 
+fn add_permission_args<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
```
Comment (ry) [inline cli/flags.rs]: add_run_args?

File: cli/flags.rs [hunk 2]
```diff
@@ -949,4 +972,28 @@ mod tests {
...
+    assert_eq!(
+      flags,
+      DenoFlags {
+        allow_net: true,
+        allow_read: true,
+        ..DenoFlags::default()
+      }
+    );
+    assert_eq!(subcommand, DenoSubcommand::Run);
+    assert_eq!(argv, svec!["deno", "script.ts"]);
```
Comment (ry) [inline cli/flags.rs]: 👍

**Comment by ry** [comment]
Working for me

**Comment by bartlomieju** [comment]
@ry please don't merge it yet, I need to correct logic for flag parsing after all

**Comment by ry** [comment]
@bartlomieju consider trying speculative url parsing

**Comment by bartlomieju** [comment]
@ry sure. I need to update logic for parsing anyway, because this PR in current state will break usages as well:
```
$ deno -r -D run --allow-net script.ts // -r -D would be ignored
```

**Comment by bartlomieju** [comment]
@ry after some thought, I am not sure if we really need to try and parse URL? 

For main module we use `root_specifier_to_url` which is a bit more relaxed than import URL parsing. If it's not a valid specifier or module is not found `main.rs::run_script` will error anyway. I don't see reason to add more logic to flag parsing to verify the URL. Thoughts?

**Comment by ry** [comment]
root_specifier_to_url is more appropriate yes

**Comment by bartlomieju** [comment]
Okay parsing logic is fixed, now both of the examples below will work properly:
```
$ deno --allow-read run --allow-net script.ts
$ deno -r -D run --allow-net script.ts
```

I also updated the manual with updated CLI usage

File: cli/flags.rs [hunk 3]
```diff
@@ -53,6 +103,7 @@ To run the REPL:
 
 To execute a sandboxed script:
 
+  deno https://deno.land/welcome.ts
   deno run https://deno.land/welcome.ts
```
Comment (ry) [inline cli/flags.rs]: I think it's better to just give one command here - this is the intro - so let's not confuse people with all the possible combinations. I suggest we use the "non-run" version for this one.

Consider updating the example on website/index.html too

**Pull request merged**